web_autoflip_demo/web_worker_pm/main.ts

```diff
@@ -0,0 +1,37 @@
+// global array buffer object to save data and pass to web worker
+declare let buffer: ArrayBuffer;
+
+// this function is called when the input loads a video
+function videoToBuffer() {
+  const file = (<HTMLInputElement>event.target).files[0];
+  console.log('video file has been chosen');
+  console.log(file);
+  const reader = new FileReader();
+  reader.onload = function () {
+    buffer = reader.result as ArrayBuffer;
+    console.log('video converted to array buffer');
+  };
+  reader.readAsArrayBuffer(file);
+}
+
+// this function is used to start a new worker
+function startWorker() {
+  const myWorker = new Worker('worker.js');
+  myWorker.postMessage(buffer);
+  console.log('passed array buffer to web worker');
+  myWorker.onmessage = function (e) {
+    console.log(e.data);
+    console.log('video received from worker');
+    // render the received video
+    renderVideo(e.data);
+  };
+}
+
+// this function is for displaying video content
+function renderVideo(videoBuffer: ArrayBuffer) {
+  const videoBlob = new Blob([videoBuffer], { type: 'video/mp4' });
+  const objectURL = URL.createObjectURL(videoBlob);
+  document.getElementById('data-vid').innerHTML =
```

Comment:
Replace string with string literal https://www.typescriptlang.org/docs/handbook/advanced-types.html#string-literal-types

Reply:
string literal also help with breaking the line so we can keep it under 80 characters

Reply:
Hi Justin, the link you introduces enum-like behavior but no general format for string literal in typescript.
I rewrite the format as my own understanding and searching online, please indicate if I did it right!

Reply:
Sorry I linked the wrong idea. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals so using `` allows you to wrap expression.

Reply:
I see and thank you for the clarification. I did use `` and please let me know if I did it right.